tests/kernel/wave/wave_e2e_test.py

```diff
@@ -1834,3 +1835,88 @@ def test(
 
     test(input, output)
     assert_close(torch_ref, output, atol=1e-03, rtol=1e-05)
+
+
+@require_e2e
+@pytest.mark.parametrize("shape", [(2, 64)])
+@param_bool("use_buffer_ops", "buf_ops")
+def test_atomic_min(shape, use_buffer_ops, request):
+    run_bench = request.config.getoption("--runperf")
+
+    M = tkl.sym.M
+    N = tkl.sym.N
+    ADDRESS_SPACE = tkl.sym.ADDRESS_SPACE
+
+    wave_size = 64
+    BLOCK_M = 2
+    BLOCK_N = 64
+    num_waves = 2
+
+    constraints: list[tkw.Constraint] = [
+        tkw.HardwareConstraint(
+            threads_per_wave=wave_size,
+            waves_per_block=(1, num_waves, 1),
+            vector_shapes={
+                M: int(BLOCK_M / num_waves),
+                N: BLOCK_N,
+            },
+        )
+    ]
+    constraints += [tkw.WorkgroupConstraint(M, BLOCK_M, 1)]
+    constraints += [tkw.WorkgroupConstraint(N, BLOCK_N, 0)]
+    constraints += [tkw.WaveConstraint(M, sympy.Integer(1))]
+    constraints += [tkw.WaveConstraint(N, BLOCK_N)]
+
+    i = tkw.IndexMapping.iterator(0)
+    j = tkw.IndexMapping.iterator(1)
+    mapping = tkw.IndexMapping(
+        num_iterators=2,
+        inputs={M: sympy.Integer(0), N: j},
+        outputs={M: sympy.Integer(0), N: j},
+    )
+    read_mapping = tkw.IndexMapping(
+        num_iterators=2, inputs={M: sympy.Integer(0), N: j}, outputs={M: i, N: j}
+    )
+
+    @tkw.wave(constraints)
+    def test(
+        a: tkl.Memory[M, N, ADDRESS_SPACE, tkl.i32],
+        c: tkl.Memory[M, N, ADDRESS_SPACE, tkl.i32],
+    ):
+        res = tkw.read(a, elements_per_thread=4)
+        # We allocate a buffer of (1,BLOCK_N) shape to perform reduction across
+        # waves. Inputs are distributed with (1,BLOCK_N) shape across each wave
+        # and performs atomic min operation on this shared memory space. Mapping
+        # attribute to atomic_min op is utilized to access the same shared memory
+        # from different waves.
+        shmem = tkw.allocate(
+            shape=(M, N),
+            distributed_shape=(1, BLOCK_N),
```

Comment:
I wonder if can infer `distributed_shape` automatically. I think logical `shape` should be `(M / 2, N)`, then, if we replace M and N with corresponding BLOCK_M, BLOCK_N we will get `distributed_shape` we want.